examples/online-shop/service.bal

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerinax/oracledb;
+
+const string USER = "balUser";
+const string PASSWORD = "balpass";
+const string HOST = "localhost";
+const int PORT = 1521;
+const string DATABASE = "ORCLCDB.localdomain";
+
+final oracledb:Client dbClient = check new(host = HOST, user = USER, password = PASSWORD, port = PORT,
+    database = DATABASE, connectionPool = {maxOpenConnections: 3, minIdleConnections: 1});
+listener http:Listener onlineShopListener = new (9090);
+
+service /onlineshop on onlineShopListener {
+    resource function get employees() returns Employee[]|error? {
+        Employee[] employees = [];
+         stream<Employee, error?> resultStream = dbClient->query(`SELECT * FROM EMPLOYEES`);
+         _ = check resultStream.forEach(function(Employee employee) {
+            employees.push(employee);
+         });
+         check resultStream.close();
+         return employees;
+    }
+
+    resource function get employees/[decimal id]() returns Employee|error? {
+        stream<Employee, error?> resultStream = dbClient->query(`SELECT * FROM EMPLOYEES WHERE EMPLOYEE_ID = ${id}`);
+        record {|Employee value;|}? data = check resultStream.next();
+        check resultStream.close();
+        return data?.value;
+    }
+
+    resource function get totalemployeesCount() returns int|error? {
+        return check dbClient->queryRow(`SELECT COUNT(*) FROM EMPLOYEES`, int);
+    }
+
+    resource function get products() returns Product[]|error? {
+        Product[] products = [];
+        stream<Product, error?> resultStream = dbClient->query(`SELECT * FROM PRODUCTS`);
+        _ = check resultStream.forEach(function(Product product) {
+           products.push(product);
+        });
+        check resultStream.close();
+        return products;
+    }
+
+    resource function get products/[decimal id]() returns Product|error? {
+        stream<Product, error?> resultStream = dbClient->query(`SELECT * FROM PRODUCTS WHERE PRODUCT_ID = ${id}`);
```

Comment:
`queryRow`?

Reply:
changed in https://github.com/ballerina-platform/module-ballerinax-oracledb/pull/36/commits/9c39326fb639c7553dd84fc179accd374e46f2ff